src/lando/api/tests/test_landings.py

```diff
@@ -777,29 +829,32 @@ def test_format_single_success_changed(
     ), "Autoformat via amending should only land a single commit."
 
 
+def _scm_get_previous_hash(scm: AbstractSCM) -> str:
+    if scm.scm_type() == SCM_TYPE_HG:
+        return HgSCM.run_hg(scm, ["log", "-r", "tip^", "-T", "{node}"]).decode("utf-8")
+    return GitSCM._git_run("rev-parse", "HEAD^", cwd=scm.path)
```

Comment:
We should have an `AbstractSCM.get_previous_hash` instead of this function which checks for the specific implementation.

Reply:
If these are only used in tests, I would hesitate to put them directly in the main implementation, but maybe they can be added in a mixin or a test class that extends `AbstractSCM`. Again, depends if they are only used in tests.

Reply:
I had the same thought as @zzzeid here. Same for the other one that gets the whole commit message.

Both are only used in tests at the moment.